psl/psl-core/src/validate/validation_pipeline/validations/relation_fields.rs

```diff
@@ -151,8 +152,21 @@ pub(super) fn referential_actions(field: RelationFieldWalker<'_>, ctx: &mut Cont
     let msg_foreign_keys = |action: ReferentialAction| {
         let allowed_actions = connector.referential_actions();
 
+        let additional_info = match (action, connector.provider_name()) {
+            (ReferentialAction::SetDefault, "mysql") => Some(formatdoc! {
+                r#"
+                    . `{set_default}` is invalid for {provider} when using `relationMode = \"foreignKeys\"`, as {provider} does not support the `SET DEFAULT` referential action.
```

Comment:
This also happens when not `relationMode` is set. Can we take that into account in the error message wording?

Also: MySQL docs say it _does_ support it. How can we make this clearer? "... does not actually support properly ..." is probably a bit too salty :D

Reply:
Help with the message wording is welcomed :)

Reply:
No, for the user there is no relationMode at all in use in most cases.

This could work:
"when using" => "(by default or when using `relationMode=foreignKeys`)"

Reply:
I find this error: "        [1;91merror[0m: [1mError validating: Invalid referential action: `SetDefault`. Allowed values: (`Cascade`, `Restrict`, `NoAction`, `SetNull`). `SetDefault` is invalid for MySQL when using `relationMode = \"foreignKeys\"`, as MySQL does not support the `SET DEFAULT` referential action."

repetitive. We already say in the first part of the message what actions are allowed, the second part just repeats that SetDefault is not supported. We add code complexity for no context added to the user.

Reply:
How about:

> Invalid referential action: `SetDefault`. Allowed values: (`Cascade`, `Restrict`, `NoAction`, `SetNull`). Starting from `Prisma v4.6.0`, we disabled `SetDefault` on MySQL (when using `relationMode = "foreignKeys"` or when not using referential integrity at all), as MySQL does not natively support the `SET DEFAULT` referential action. Learn more at ..."

Reply:
that's fine

Reply:
We should not include version information in error messages.

Reply:
The second bit of the error message is because we differ from what MySQL supports there @tomhoule. People can have this in their database, and after Introspection they will get this and then get a validation error. This second sentence gives the necessary context to be able to understand and then links to documentation that goes deeper into that.

Reply:
(We'll talk about this in our team sync)